### PR TITLE
Python 3.7 Compatibility

### PR DESCRIPTION
**Pull request by ravi-mosaicml**
1. Fallbacking to `Protocol = object` in 3.7
2. Fallbacking to `final = lambda x: x` in python 3.7
3. Removed custom warning type for DeferredLogMetricWarning since pytest filterwarnigns errored

### CODE REVIEW

**Comment by hanlint** [comment]
Before merging this, can we run some testing with some simple examples on colab?

**Comment by hanlint** [comment]
A few other requests:
* update the docs/README where we mention requirements
* add something bout colab after it tested?

File: setup.py
Comment (Averylamp) [inline setup.py:30]: Don't worry about changing Yahp versions, I'll handle that separately just so I can ensure to track it properly

Comment (Averylamp) [reply setup.py:30]: Otherwise looks good to me, but ditto on hanlins requests

Comment (ravi-mosaicml) [reply setup.py:30]:
```suggestion
    "yahp>=0.0.10",
```

Comment (ravi-mosaicml) [reply setup.py:30]: I'll keep it here just so we don't merge this in until new yahp is released (this diff should disappear once you add that commit in).

Comment (Averylamp) [reply setup.py:30]: Yeah, fyi YAHP is already on `0.0.11`, so that isn't the right number to ensure it updates with YAHP either.  I think I'd rather just track all version numbers independently for now to keep it lower overhead for me.

Comment (Averylamp) [reply setup.py:30]: Marked with a new label, but as this is a backwards compatible change you should be fine to merge into dev

**Review by Landanjs — COMMENTED** [review]
I can't speak to the code, but it looks like the examples (MNIST and CIFAR) are trainingin [collab](https://colab.research.google.com/drive/1HscE_39ywgGZhI2glt-jF_PNElqh4Pez?usp=sharing)!

Could someone look at the notebook and see if anything else should be tested?

**Review by Averylamp — APPROVED** [review]
lgtm

**Pull request merged**
